WordPress/Classes/Services/AuthenticationService.swift

```diff
@@ -94,7 +94,9 @@ class AuthenticationService {
                     // so that the user sees a reasonable error situation on screen.
                     // We could opt to create a special screen but for now I'd rather users report
                     // the issue when it happens.
-                    failure(error)
+                    DispatchQueue.main.async {
+                        failure(error)
+                    }
```

Comment:
`success()` was called on main thread while `failure()` was called on background thread. This is for consistency.

Reply:
I'm wondering which success was called.  The one on line 81 (which would be main thread) or the one on line 87 (which doesn't look like it would be the main thread).  
I agree we should dispatch to the main thread but I wonder if we could move this into the `requestAuthCookies` method and wrap both `success` and `failure` calls there so its all done in one place? 
(I coincidentally ran across the threading issue while putting together #14363 and remembered you'd made this change.)

Reply:
Moving it to `requestAuthCookies` sounds good!

For `success` on line 87, execution has been already moved to main thread by `setCookies`. At least in the `WKHTTPCookiesStore` implementation of it. That's why it works.

But I agree this is shady, so let's improve :)